kubernetes/eventhandler.go

```diff
@@ -214,10 +239,20 @@ func (n *nodePodUpdater) OnUpdate(obj interface{}) {
 		n.locker.Lock()
 		defer n.locker.Unlock()
 	}
-	for _, pod := range n.store.List() {
-		pod, ok := pod.(*Pod)
-		if ok && pod.Spec.NodeName == node.Name {
-			n.handler(pod)
+	oldobject := n.nodewatcher.Oldobject()
```

Comment:
here, cachedObject and cachedNode instead

Reply:
Done in https://github.com/elastic/elastic-agent-autodiscover/pull/71/commits/16b0fca86b19f8f8f108c219f43f1024687da2ca